File: java/src/main/java/fr/sdecout/handson/rest/library/LibrarySearchResponseItem.java
package fr.sdecout.handson.rest.library;

import com.fasterxml.jackson.annotation.JsonUnwrapped;
import fr.sdecout.handson.rest.shared.LibraryField;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public record LibrarySearchResponseItem(
        @JsonUnwrapped @NotNull @Valid LibraryField library
) {
}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/library/LibrarySearchResponseItem.kt
package fr.sdecout.handson.rest.library

import fr.sdecout.handson.rest.shared.AddressField
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class LibrarySearchResponseItem(
    @field:NotBlank val id: String?,
    @field:NotBlank val name: String?,
    @field:NotNull @field:Valid val address: AddressField?,
)
